### PR TITLE
Fix Python SyntaxWarnings generated by compiled code

### LINKED ISSUE #996: `basilisp.core/nil?` SyntaxWarning with numeric constants
Hi,

starting with Python 3.8, using `basilisp.core/nil?` with a numeric literal triggers a SyntaxWarning:
```
SyntaxWarning: "is" with 'int' literal. Did you mean "=="?
```

To reproduce, open up the REPL and call `nil?` with a numeric argument, you should see the warning
``` clojure
> poetry run basilisp repl
basilisp.user=> (nil? 4)
<REPL Input>:1: SyntaxWarning: "is" with 'int' literal. Did you mean "=="?
false
```

This warning also appears during running the test suite, as some macro-expanded code includes nil? with a literal as its first argument. For example:

```
tests\basilisp\test_core_macros.lpy:598
  C:\src\basilisp\tests\basilisp\test_core_macros.lpy:598: SyntaxWarning: "is" with 'int' literal. Did you mean "=="?
    (is (= 2 (some-> 1 inc)))
```

The underlying issue is that `basilisp.core/nil?` is defined using Python’s is operator
```clojure
(defn ^:inline nil?
  "Return ``true`` if ``x`` is ``nil``\\, otherwise ``false``\\."
  [x]
  (operator/is- x nil))
```

However, the is operator should not be used with string or numeric literals, as outlined in https://bugs.python.org/issue34850

> Gregory have noticed that the "is" and "is not" operator sometimes is used with string and numerical literals. This code "works" on CPython by accident, because of caching on different levels (small integers and strings caches, interned strings, deduplicating constants at compile time). But it shouldn't work on other implementations, and can not work even on early or future CPython versions.

and made it into a SyntaxWarning since Python 3.8.

Although comparing a literal with nil/None (as we do in this function) might not cause problems, I believe the warning needs to be addressed.

A solution can be to use the `==` operator instead in `nil?`,  but [PEP-8](https://peps.python.org/pep-0008/#programming-recommendations) suggests comparison to `None` should only be done with `is`:

>Comparisons to singletons like None should always be done with is or is not, never the equality operators.

Perhaps a better solution could be to implement `nil? in the compiler, where the analyzer can check if the argument is a literal and decide accordingly?

Thanks

### PR DESCRIPTION
**Pull request by chrisrink10**
Fixes #996

### CODE REVIEW

File: src/basilisp/core.lpy [hunk 1]
```diff
@@ -1606,7 +1606,7 @@
   [x]
   (python/int x))
 
-(defn ^:inline boolean
+(defn ^:no-inline boolean
```
Comment (chrisrink10) [inline src/basilisp/core.lpy:1609]: When this function is inlined with the constants `0` and `0.0`, the generated code for calls to `false?` becomes `False == 0` and `False == 0.0` respectively with the changes in this PR. In Python, those expressions are always true. The way we distinguish between them for Clojure compatibility (since only `nil` and `false` are logical false in Clojure) is to use `is`, but when using inlined constants that generates a syntax warning.

The easiest fix is to just not inline the function call since using `is` on a variable (even one containing `0` or `0.0`) is legal syntax.

File: src/basilisp/lang/util.py [hunk 2]
```diff
@@ -14,6 +14,7 @@
 _DOUBLE_DOT_REPLACEMENT = "__DOT_DOT__"
 
 _MUNGE_REPLACEMENTS = {
+    "'": "__PRIME__",
```
Comment (chrisrink10) [inline src/basilisp/lang/util.py:17]: This was an unrelated change after PR #924 merged since it uses the `v'` nomenclature for a variable which was not previously covered by `munge`. I missed this change in #650.
